File: backend/src/CoinAlertApi/Domain/Entities/Opportunity.cs
using CoinAlertApi.Domain.Enums;
using CoinAlertApi.Domain.Interfaces;

namespace CoinAlertApi.Domain.Entities;

public class Opportunity : IEntity
{
    public string Id { get; set; } = string.Empty;
    public string CryptoId { get; set; } = string.Empty;
    public OpportunityType Type { get; set; }
    public decimal TargetPrice { get; set; }
    public decimal CurrentPrice { get; set; }
    public OpportunityStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? TriggeredAt { get; set; }
}

File: frontend/coin-alert/src/types/opportunity.ts
export enum OpportunityType {
  Buy = 0,
  Sell = 1,
}

export enum OpportunityStatus {
  Active = 0,
  Triggered = 1,
}

export interface Opportunity {
  id: string
  cryptoId: string
  type: OpportunityType
  targetPrice: number
  currentPrice: number
  status: OpportunityStatus
  createdAt: string
  triggeredAt: string | null
}

export interface CreateOpportunityDto {
  cryptoId: string
  type: OpportunityType
  targetPrice: number
}

export interface OpportunityTriggeredEvent {
  id: string
  cryptoId: string
  type: OpportunityType
  targetPrice: number
  currentPrice: number
  triggeredAt: string
}
